r2/r2/lib/media.py

```diff
@@ -312,6 +312,16 @@ def _scrape_media(url, autoplay=False, maxwidth=600, force=False,
             print "%s made a bad secure media obj for url %s" % (scraper, url)
             secure_media_object = None
 
+        # If thumbnail can't be found, attempt again using _ThumbnailOnlyScraper
+        # This should fix bugs that occur when embed.ly caches links before the thumbnail is available
+        if not thumbnail_image:
```

Comment:
So I think this probably isn't quite the correct approach, since even if the `_ThumbnailOnlyScraper` was chosen initially (for a site embedly doesn't support), this is going to end up just trying again with exactly the same method if it didn't find a thumbnail the first time.

You could probably either try to move this into the "didn't find anything" case inside `_EmbedlyScraper` itself - probably here: https://github.com/reddit/reddit/blob/master/r2/r2/lib/media.py#L682

Or alternatively if you think it makes sense to be out here as a general fallback for *all* scrapers, it should probably just check to make sure it's not repeating, something like `if not thumbnail_image and not isinstance(scraper, _ThumbnailOnlyScraper):`